Source/monster.cpp

```diff
@@ -4698,19 +4698,19 @@ bool DirOK(int i, Direction mdir)
 
 	commitment((DWORD)i < MAXMONSTERS, i);
 	Point futurePosition = monster[i].position.tile + mdir;
-	if (futurePosition.y < 0 || futurePosition.y >= MAXDUNY || futurePosition.x < 0 || futurePosition.x >= MAXDUNX || !PosOkMonst(i, futurePosition.x, futurePosition.y))
+	if (futurePosition.y < 0 || futurePosition.y >= MAXDUNY || futurePosition.x < 0 || futurePosition.x >= MAXDUNX || !PosOkMonst(i, futurePosition))
 		return false;
 	if (mdir == DIR_E) {
-		if (SolidLoc(futurePosition.x, futurePosition.y + 1) || dFlags[futurePosition.x][futurePosition.y + 1] & BFLAG_MONSTLR)
+		if (SolidLoc(futurePosition + Point { 0, 1 }) || dFlags[futurePosition.x][futurePosition.y + 1] & BFLAG_MONSTLR)

```

Comment:
Thease are checking if the player is diagonally blocked by a monster so it would actually make sens to use the DIR_* for checking the relative position.

If `monster` tries to move `DIR_E`, check for monster at `DIR_E+DIR_SW`. But even better would be to not add mdir and the using an illogically direction that subtraction it.

```suggestion
		if (SolidLoc(position + DIR_SE) || dFlags[position.x + 1][futurePosition.y] & BFLAG_MONSTLR)
```

Reply:
I was sure these additions were direction-based but I didn't spend any time looking into the logic itself: this refactor was more of a "dumb" replacement and then I'd go back to these and refactor the logic.

Thanks for explaining it, I'll apply now in this case.